### PR TITLE
BUG: fix np.load() of empty .npz file

### LINKED ISSUE #9989: np.load() fails to read empty archive created by np.savez()
Try this:

    np.savez('nothing.npz') # OK
    np.load('nothing.npz') # error

Loading fails:

    OSError: Failed to interpret file 'nothing.npz' as a pickle

It should succeed (and give you an object with no arrays inside).  Likewise for `savez_compressed()` (which also currently fails).

I'm using NumPy 1.13.1.

### PR DESCRIPTION
**Pull request by jzwinck**
Fixes #9989

### CODE REVIEW

**Comment by mattip** [comment]
Trivial test missing for `savez_compressed()`, just to be complete`. 

Should we prevent creating such files during savez()? Was there any discussion around that idea?

**Comment by jzwinck** [comment]
@mattip There is no need for a test using `savez_compressed()`, because (1) the fixed bug is in `load()`, and (2) an empty Zip file doesn't really have a compression level, so `savez_compressed()` is exactly the same as `savez()` when no arrays are stored.

As for preventing such files during `savez()`, I don't prefer that because it breaks backward compatibility.  Some people might want to save empty .npz files, and I haven't seen anyone complain that `savez()` works with no arrays.

File: numpy/lib/npyio.py [hunk 1]
```diff
@@ -412,12 +412,13 @@ def load(file, mmap_mode=None, allow_pickle=True, fix_imports=True,
...
         # Code to distinguish from NumPy binary files and pickles.
         _ZIP_PREFIX = b'PK\x03\x04'
+        _ZIP_SUFFIX = b'PK\x05\x06' # empty zip files start with this
         N = len(format.MAGIC_PREFIX)
         magic = fid.read(N)
         # If the file size is less than N, we need to make sure not
         # to seek past the beginning of the file
         fid.seek(-min(N, len(magic)), 1)  # back-up
-        if magic.startswith(_ZIP_PREFIX):
+        if magic.startswith(_ZIP_PREFIX) or magic.startswith(_ZIP_SUFFIX):
```
[the remark this answers is not in the document]
Comment (jzwinck) [reply numpy/lib/npyio.py]: @tylerjereddy Loading an empty text file is fundamentally not the same as loading an empty archive.  Note that `np.load()` of a `npy` file containing a single empty array does not warn.

**Comment by jzwinck** [comment]
@mattip What do you think about merging this?

**Comment by juliantaylor** [comment]
looks good, thanks

**Pull request merged**